lang/stata/stata.py

```diff
@@ -0,0 +1,157 @@
+from talon import Context, actions, settings
+
+ctx = Context()
+
+ctx.matches = r"""
+code.language: stata
+"""
+
+# functions.py
+ctx.lists["user.code_parameter_name"] = {
+    # regressions
+    "V C E cluster": "vce(cluster)",
+    "V C E robust": "vce(robust)",
+}
+
+# functions_common.py
+ctx.lists["user.code_common_function"] = {
+    # base stata
+    "global": "global",
+    "local": "local",
+    "reg": "reg",
+    "regress": "reg",
+    # packages
+    "estadd": "estadd",
+    "estout": "estout",
+    "estpost": "estpost",
+    "eststo": "eststo",
+    "esttab": "esttab",
+}
+
+# libraries_gui.py
+ctx.lists["user.code_libraries"] = {
+    "estout": "estout",
+}
+
+
+@ctx.action_class("user")
+class UserActions:
+    # comment_line.py
+    def code_comment_line_prefix():
+        actions.auto_insert("* ")
+
+    # functions.py
+    def code_private_function(text: str):
+        result = "program {} \n\nend".format(
+            actions.user.formatted_text(
+                text, settings.get("user.code_private_function_formatter")
+            )
+        )
+        actions.user.paste(result)
+        actions.edit.up()
+        actions.key("tab")
+
+    def code_default_function(text: str):
+        actions.user.code_private_function(text)
+
+    def code_insert_named_argument(parameter_name: str):
+        actions.insert(f"{parameter_name} ")
+
+    # functions_common.py
+    def code_insert_function(text: str, selection: str):
+        text += f" {selection or ''}"
+        actions.user.paste(text)
+
+    # imperative.py
+    def code_block():
+        actions.auto_insert("\n")
+
+    def code_state_if():
+        actions.insert("if  {\n\n}\n")
+        actions.key("up:2 tab up left")
+
+    def code_state_else_if():
+        actions.insert("else if  {\n\n}\n")
+        actions.key("up:2 tab up right:4")
+
+    def code_state_else():
+        actions.insert("else {\n\n}\n")
+        actions.key("up:2 tab")
+
+    def code_state_for():
+        actions.insert("forval  {\n\n}\n")
+        actions.key("up:2 tab up right:3")
```

Comment:
is it possible people would/could configure the stada app to not use 4 space tabs? if so I guess this would break. same goes for the other spots you added tab

Reply:
Per default stata inserts 4 spaces per tab. However, this can be configured in the editor. But I guess if someone configured this, talon actually does press tab and this would insert the modified number of spaces, correct?

What would be the alternative? Not adding the tab? `\t`?

Reply:
Ya, that's my worry if someone configures it differently than the default then it will be wrong. I think the only option is to just manually press the keys unfortunately, which will be a bit slower for the user, but at least correct. I believe `\t` will be the same as pressing tab.

Reply:
Ah, Now I understand.  Will make these changes either tonight or tomorrow.

Reply:
I have made the according changes.  The code should now be robust to how many spaces are defined as a tab.

 However, if one works in an editor with auto closing brackets on, one would still run into problems. But I guess this affects all languages in talon.  one could create something like context sensitive dictation I guess. But this goes beyond the scope of this pull request.